### ISSUE TITLE
Autocomplete of Literal value only takes last overload into account

### ISSUE THREAD

**Issue by friedererdmann** [microsoft/pyright#5193]
**Describe the bug**
In our code base we have a function which returns different objects based on a string provided as parameter. To provide good auto complete, our stubs make use of Literal and overload.

![image](https://github.com/microsoft/pyright/assets/55715326/d8aa8c11-852c-46d6-b3a5-d93226615c14)

**To Reproduce**
Create a function `example` and overload it with different `Literal` values. When calling the function now you can start typing quotation marks to see the possible strings to type, but you'll only be offered the last possible overload.

**Expected behavior**
All overloads should be offered to the autocompletion

**Screenshots or Code**
```python
from typing import overload, Literal


@overload
def example(parameter: Literal["int"]) -> int: ...


@overload
def example(parameter: Literal["bool"]) -> bool: ...


@overload
def example(parameter: Literal["str"]) -> str: ...


def example(parameter):
    # some logic here
    return parameter


example("")
```

**VS Code extension or command-line**
Using PyLance as a VSCode extension (which includes Pyright (?))

**Comment by friedererdmann** [comment]
Actually found a work around by making one final overload:
```python
@overload
def example(parameter: Literal["int", "bool", "str"]): ...
```

**Comment by erictraut** [comment]
The completion provider is working as designed here, so I don't consider this a bug. However, I think this is a reasonable enhancement request. Since this isn't a core type checking feature, it would be a feature request for the pylance team.

@rchiodo, could you please transfer this to the pylance-release project and consider it as a potential future enhancement request for the completion provider?
